src/vm/gcenv.os.cpp

```diff
@@ -752,25 +789,83 @@ bool GCToOSInterface::CanEnableGCNumaAware()
     return NumaNodeInfo::CanEnableGCNumaAware() != FALSE;
 }
 
-bool GCToOSInterface::GetNumaProcessorNode(PPROCESSOR_NUMBER proc_no, uint16_t *node_no)
+bool GCToOSInterface::GetNumaProcessorNode(uint16_t proc_no, uint16_t *node_no)
 {
     LIMITED_METHOD_CONTRACT;
 
-    return NumaNodeInfo::GetNumaProcessorNodeEx(proc_no, node_no) != FALSE;
-}
+    GroupProcNo groupProcNo(proc_no);
 
-bool GCToOSInterface::CanEnableGCCPUGroups()
-{
-    LIMITED_METHOD_CONTRACT;
+    PROCESSOR_NUMBER procNumber;
+    procNumber.Group    = groupProcNo.GetGroup();
+    procNumber.Number   = (BYTE)groupProcNo.GetProcIndex();
+    procNumber.Reserved = 0;
 
-    return CPUGroupInfo::CanEnableGCCPUGroups() != FALSE;
+    return NumaNodeInfo::GetNumaProcessorNodeEx(&procNumber, node_no) != FALSE;
 }
 
-void GCToOSInterface::GetGroupForProcessor(uint16_t processor_number, uint16_t* group_number, uint16_t* group_processor_number)
+// Get processor number and optionally its NUMA node number for the specified heap number
+// Parameters:
+//  heap_number - heap number to get the result for
+//  proc_no     - set to the selected processor number
+//  node_no     - set to the NUMA node of the selected processor or to NUMA_NODE_UNDEFINED
+// Return:
+//  true if it succeeded
+bool GCToOSInterface::GetProcessorForHeap(uint16_t heap_number, uint16_t* proc_no, uint16_t* node_no)
 {
-    LIMITED_METHOD_CONTRACT;
+    bool success = false;
 
-    return CPUGroupInfo::GetGroupForProcessor(processor_number, group_number, group_processor_number);
+    if (CPUGroupInfo::CanEnableGCCPUGroups())
+    {
+        uint16_t gn, gpn;
+        CPUGroupInfo::GetGroupForProcessor((uint16_t)heap_number, &gn, &gpn);
+
+//      dprintf(3, ("using processor group %d, mask %Ix for heap %d\n", gn, (uintptr_t)1 << gpn), heap_number));
```

Comment:
>dprintf [](start = 8, length = 7)

seeing that you cannot use dprintf's here, the lines with dprintf's should be deleted

Reply:
I have left it there just to see if you'd think this dprintf was important enough so that we should put it to the caller side and print similar thing based on the values returned by the GetProcessorForHeap.